cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/VersionInfo.java

```diff
@@ -23,4 +23,83 @@
  */
 public interface VersionInfo extends Comparable<Object> {
 
+  /**
+   * @return major version by parsing the versionInfo string,
+   * in the format "major.minor.fix-buildTime"/"major.minor.fix-SNAPSHOT-buildTime"
+   */
+  default int getMajor() {
+    String[] versionInfo = this.toString().split("-");
+    if (versionInfo.length > 0) {
+      int idx = versionInfo[0].indexOf('.');
+      if (idx > 0) {
+        return Integer.parseInt(versionInfo[0].substring(0, idx));
+      }
+    }
+    return 0;
+  }
+
+  /**
+   * @return minor version by parsing the versionInfo string,
```

Comment:
The javadoc for the interface shouldn't talk about the implementation (like parsing string) directly. It should be something like:

```
Gets the minor version part of the version string. The default implementation is to parse the version string....
@return the minor version
```

Reply:
ack